tyk-pro/templates/deployment-gw-repset.yaml

```diff
@@ -97,8 +106,15 @@ spec:
             value: "{{ .Values.secrets.APISecret }}"
           - name: TYK_GW_NODESECRET
             value: "{{ .Values.secrets.APISecret }}"
+        # Enable backward compatibility to chart 0.8.1.
+        {{- if or .Values.gateway.sharding .Values.enableSharding}}
           - name: TYK_GW_DBAPPCONFOPTIONS_TAGS
+          {{- if .Values.gateway.sharding }}
```

Comment:
It's better to check whether `.Values.gateway.sharding.tags` is empty or not and based on that condition you can insert tags.

Please check others and follow this way.

Reply:
Agreed.